src/wizards/tokenSwapDealWizard/stages/termsStage/termClause/termClause.ts

```diff
@@ -80,6 +108,13 @@ export class TermClause {
             return;
           }
           const viewContent = marked(textContent);
+          if (textContent.length + editor.getData().length >= 500) {
```

Comment:
Is it possible to move this disable logic into the editor?
I don't see why it should go in `termClause.ts`.
I would just want to configure the editor like `option: {limit: 500}`.

If not possible, could you give a reason why the way you did it is "the only way"?